src/Gateways/GpApiGateway.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace GlobalPayments\WooCommercePaymentGatewayProvider\Gateways;
+
+use GlobalPayments\Api\Entities\Enums\Environment;
+use GlobalPayments\Api\Entities\Enums\GatewayProvider;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Gateways\ThreeDSecure\CheckEnrollment;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Gateways\ThreeDSecure\GpApiGateway_Webhook_Handler;
+use GlobalPayments\WooCommercePaymentGatewayProvider\Plugin;
+use WC_Order;
+
+defined( 'ABSPATH' ) || exit;
+
+class GpApiGateway extends AbstractGateway {
+    /**
+     * Defines production environment
+     */
+    const ENVIRONMENT_PRODUCTION = 'production';
+
+    /**
+     * Defines sandbox environment
+     */
+    const ENVIRONMENT_SANDBOX    = 'sandbox';
+
+    /**
+     * SDK gateway provider
+     *
+     * @var string
+     */
+    public $gateway_provider = GatewayProvider::GP_API;
+
+    /**
+	 * App ID
+	 *
+	 * @var string
+	 */
+	public $app_id;
+
+	/**
+	 * App Key
+	 *
+	 *
+	 * @var string
+	 */
+	public $app_key;
+
+	/**
+	 * Should live payments be accepted
+	 *
+	 * @var bool
+	 */
+	public $is_production;
+
+	/**
+	 * Integration's Developer ID
+	 *
+	 * @var string
+	 */
+	public $developer_id = '';
+
+	public function configure_method_settings() {
+		$this->id                 = 'globalpayments_gpapi';
+		$this->method_title       = __( 'GP-API', 'globalpayments-gateway-provider-for-woocommerce' );
+		$this->method_description = __( 'Connect to the Global Payments API (GP-API) gateway', 'globalpayments-gateway-provider-for-woocommerce' );
+	}
+
+	public function get_first_line_support_email() {
+		return 'securesubmitcert@e-hps.com';
```

Comment:
This should be updated to the correct support email for GP API.

Reply:
Waiting for Colleen to provide one.